lib/active_merchant/billing/gateways/agms.rb

```diff
@@ -0,0 +1,322 @@
+        # Invoice Data
+        add_invoice(post, money, options)
+        # Add Authorization
+        add_authorization(post, authorization, options)
+        # Call the commit function to pass on the params to gateway
+        commit('refund', post)
+      end
+
+      def void(authorization, options={})
+        # Void transaction, we need to pass on only authorization number for voiding transaction
+        post = {}
+        # Add Authorization
+        add_authorization(post, authorization, options)
+        # Call the commit function to pass on the params to gateway
+        commit('void', post)
+      end
+
+      def verify(credit_card, options={})
+        # Verify is two step process, first we fire the authorize then we capture the auth and issue void
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        # Indicates whether before dumping the params it remove the sensitive information
+        true
+      end
+
+      def scrub(transcript)
+        # Scrubbing logic, we remove CCNumber, CVV, GatewayUserName and GatewayPassword before it is dumped
+        # The module also support dumping of all the request response in file for debugging and stubbing, this ensure no sensitive info is dumped
+        transcript.
+          gsub(%r((<CCNumber>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<CVV>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<GatewayUserName>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<GatewayPassword>)[^<]+(<))i, '\1[FILTERED]\2')
+      end
+
+      # Private methods
+      private
+
+      def add_payment(post, payment)
+        # Add payment, by default we deal with card only and assign payment to card data
+        creditcard = payment
+        post[:CCNumber]  = creditcard.number
+        post[:CVV] = creditcard.verification_value if creditcard.verification_value?
+        post[:CCExpDate]  = expdate(creditcard)
+        post[:FirstName] = creditcard.first_name
+        post[:LastName]  = creditcard.last_name   
+      end
+
+      def add_authorization(post, authorization, options)
+        # Add the authorization code
+        post[:TransactionID] = authorization
+      end 
+
+      def add_invoice(post, money, options)
+        # Add invoice, here money can be in multiple currency but since we deal with only USD we assign it directly, refer active_merchant docs
+        post[:Amount] = amount(money)
+        if options.has_key? :order_id
+          post[:OrderID] = options[:order_id]
+        end
+        if options.has_key? :description
+          post[:OrderDescription] = options[:description]
+        end
+        if options.has_key? :invoice
+          post[:PONumber] = options[:invoice]
+        end
+        # Agms support only USD so we do not pass though currency convertor function
```

Comment:
It's fine to not support `:currency`, you can just delete these comments/commented code.